### PR TITLE
coredump_log: coredump log in panic mode

### PR DESCRIPTION
**Pull request by RobertGalatNordic**
Current implementation assumes that CONFIG_LOG_IMMEDIATE=y
guarantees complete transfer, and it is not true.
In my opinion, core dump should always be printed in panic mode.

Signed-off-by: Robert Gałat <robert.galat@nordicsemi.no>

### CODE REVIEW

**Review by dcpleung — APPROVED** [review]

**Review by chen-png — APPROVED** [review]

**Comment by RobertGalatNordic** [comment]
I can see some approvals, but I can not merge the change because some workflows await for approval. @dcpleung As you approved the change, please feel free to merge the change

**Comment by dcpleung** [comment]
Need to fix the compliance failure.

**Comment by cfriedt** [comment]
@nashif - please review when you have some time.

**Pull request merged**
